### PR TITLE
Vstest.console should exit on parent process exit in design mode

### PR DESCRIPTION
**Pull request by smadala**
**User scenario:** Close VS while test discovery is in progress, vstest.console and it's sub process becomes orphans processes whose holds locks on adapters and test dlls. This causes project build fail on subsequent run.

**Root cause:** Vstest.console main thread was busy in discovery which won't receive EndSession from VS.
Parent process exit call back just disposes communication channel.
**Fix:** Make discovery request async and parent process exit call back should call `Environment.Exit(1)`.
**Bug:** [DevDiv bug](https://devdiv.visualstudio.com/DefaultCollection/DevDiv/VS.in%20Agile%20Testing%20IDE/_workitems?id=450161&triage=true&fullScreen=true&_a=edit)

### CODE REVIEW

**Review by Faizan2304 — APPROVED** [review]
LGTM 👍

**Comment by Faizan2304** [comment]
@dotnet-bot test Windows_NT / Debug Build please

File: test/Microsoft.TestPlatform.Client.UnitTests/DesignMode/DesignModeClientTests.cs [hunk 1]
```diff
@@ -321,5 +322,51 @@ public void InvokeCustomHostLaunchAckCallback(int processId, string errorMessage
                     new Message() { MessageType = MessageType.CustomTestHostLaunchCallback, Payload = JToken.FromObject(payload) });
             }
         }
+
+        [TestMethod]
+        public void DesignModeClientConnectShouldSendTestMessageAndDiscoverCompleteOnExceptionInDiscovery()
+        {
+            var complateEvent = new AutoResetEvent(false);
```
Comment (codito) [inline test/Microsoft.TestPlatform.Client.UnitTests/DesignMode/DesignModeClientTests.cs]: nit: public methods should move up. Can the setup code be extracted?

File: test/vstest.console.UnitTests/Processors/PortArgumentProcessorTests.cs [hunk 2]
```diff
@@ -51,7 +64,7 @@ public void CapabilitiesShouldAppropriateProperties()
         [TestMethod]
         public void ExecutorInitializeWithNullOrEmptyPortShouldThrowCommandLineException()
         {
-            var executor = new PortArgumentExecutor(CommandLineOptions.Instance, TestRequestManager.Instance);
+            var executor = new PortArgumentExecutor(CommandLineOptions.Instance, this.testRequestManager.Object);
```
Comment (codito) [inline test/vstest.console.UnitTests/Processors/PortArgumentProcessorTests.cs]: Make it instance variable?

**Review by codito — APPROVED** [review]

**Comment by smadala** [comment]
@dotnet-bot test Windows_NT / Debug Build please

**Comment by smadala** [comment]
@dotnet-bot test Windows_NT / Release Build please

**Comment by smadala** [comment]
@dotnet-bot test Windows_NT / Debug Build please

**Pull request merged**